### PR TITLE
Antlr: reject && at the end + refactor grammar #134

### LINKED ISSUE #134: Reject && at the end of cryptarithms and multiple && side by side
1. The parser should not accept a cryptarithm having the && operator at the end of it.
  Example:
  `a + b = c &&` -> should be rejected
  `a + b = c;` -> should be accepted

2. The parser should not accept multiple && side by side
   Example:
   `a + b = c &&&& a + c = d` -> should be rejected

### PR DESCRIPTION
**Pull request by FissoreD**
- close #134

### CODE REVIEW

File: src/main/antlr4/Cryptator.g4 [hunk 1]
```diff
@@ -10,65 +10,60 @@
...
 SYMBOL : [a-zA-Z\u0080-\uFFFF] {};
+NUMBER : [0-9]+ {};
 
-DIGIT : [0-9] {};
-
-AND : ';' | '&&';
-
+// CONJUNCTIONS
+AND : ';';
+AND_INFIX : '&&';
```
Comment (arnaud-m) [inline src/main/antlr4/Cryptator.g4:63]: - `;` statements separator
- `&&` logical conjunction

Rename `AND` to `SC` and `AND_INFIX` to `AND` ?

File: src/test/java/cryptator/SolverTest.java [hunk 2]
```diff
@@ -492,7 +489,7 @@ public void testSendMoreMoneyList3() throws CryptaParserException, CryptaModelEx
     public void testUnsatSendMoreMoneyList() throws CryptaParserException, CryptaModelException, CryptaSolverException {
         t.config.setHornerScheme(true);
         t.testUNSAT("send+more=money; s+e=n", "send+more=money;;; s+e=n", "send+more=money; s+e=n;",
-                "send+more=money&& s+e=n", "send+more=money&&&&&& s+e=n", "send+more=money&& s+e=n&&");
+                "send+more=money&& s+e=n", "send+more=money&&&&&& s+e=n");
```
Comment (arnaud-m) [inline src/test/java/cryptator/SolverTest.java:492]: The cryptarithm `send+more=money&&&&&& s+e=n` should raise an exception.

However, `send+more=money ;; s+e=n ;` should not ?

**Review by arnaud-m — CHANGES_REQUESTED** [review]

File: src/main/antlr4/Cryptator.g4 (hunk 1 above)
Comment (FissoreD) [reply src/main/antlr4/Cryptator.g4:63]: Sorry, what's the meaning of `SC` ?

File: src/test/java/cryptator/SolverTest.java (hunk 2 above)
Comment (FissoreD) [reply src/test/java/cryptator/SolverTest.java:492]: Ah, we don't accept multiple `&&` side by side, I missed it. I will correct this.

File: src/main/antlr4/Cryptator.g4 (hunk 1 above)
Comment (arnaud-m) [reply src/main/antlr4/Cryptator.g4]: `SC` stands for SemiColon. A better idea ?

**Review by arnaud-m — APPROVED** [review]

**Pull request merged**